File: spark-java/spark-dataframe/dataframe-column/src/main/java/com/mahfooz/df/columns/rename/RenamingColumn.java
package com.mahfooz.df.columns.rename;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class RenamingColumn {
    public static void main(String[] args) {
        SparkSession spark = SparkSession
                .builder()
                .master("local[*]")
                .appName("RenamingColumn")
                .getOrCreate();

        String dataHome = System.getenv("DATA_HOME");

        Dataset<Row> df = spark.read()
                .option("header", "true")
                .option("inferSchema", "true")
                .csv(dataHome+"\\FileData\\Csv\\Retail\\all\\online-retail-dataset.csv");

        df.printSchema();
        Dataset<Row> renamedDF = df.withColumnRenamed("InvoiceNo", "InvoiceNumber");
        renamedDF.show();

    }
}

File: spark-scala/spark-dataframe/dataframe-column/src/main/scala/com/mahfooz/dataframe/columns/rename/RenamingColumn.scala
package com.mahfooz.dataframe.columns.rename

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions.lit

object RenamingColumn {
  def main(args: Array[String]): Unit = {

    val spark = SparkSession.builder
      .appName(RenamingColumn.getClass.getSimpleName)
      .master("local")
      .getOrCreate()

    val df = spark.read
      .format("json")
      .load(args(0))

    df.withColumn("numberOne", lit(1))
      .show(2)

    df.withColumnRenamed("DEST_COUNTRY_NAME", "dest")
      .show()

  }
}
